src/Domain/LeanCode.DomainModels.EF/EFRepository.cs

```diff
@@ -9,19 +9,20 @@
 
 namespace LeanCode.DomainModels.EF
 {
-    public abstract class EFRepository<TEntity, TIdentity, TContext, TUnitOfWork> : IRepository<TEntity, TIdentity>
+    public abstract class EFRepository<TEntity, TIdentity, TContext> : IRepository<TEntity, TIdentity>
         where TEntity : class, IAggregateRootWithoutOptimisticConcurrency<TIdentity>
         where TIdentity : notnull
         where TContext : notnull, DbContext
-        where TUnitOfWork : notnull, IUnitOfWork
     {
+        protected TContext DbContext { get; }
         protected DbSet<TEntity> DbSet { get; }
-        protected TUnitOfWork UnitOfWork { get; }
+        protected Task SaveChangesAsync(CancellationToken cancellationToken = default) =>
```

Comment:
Empty line before this one please.
```suggestion

        protected Task SaveChangesAsync(CancellationToken cancellationToken = default) =>
```

Reply:
Aaand let's move it after `public` methods.